src/entities/shadow.py

```diff
@@ -2,7 +2,7 @@
 
 import logging
 import random
-from collections import Sequence
+from collections.abc import Sequence
```

Comment:
Thanks for your contribute! The best fix that compatibles with most versions of Python 3 is

```python
from typing import Sequence
```